### PR TITLE
add options to zephyr builder

### PR DESCRIPTION
**Pull request by dsemenets**
work_dir option allows build out-of-tree zephyr applicaton target option made optional.

### CODE REVIEW

**Comment by oleksiimoisieiev** [comment]
+1

**Review by lorc — COMMENTED** [review]
Would you please describe problem that you are trying to fix with this patch? Please do this in the commit message.

**Comment by lorc** [comment]
Also, don't forget to update corresponding documentation. In the same commit.

**Comment by dsemenets** [comment]
> As I understand this change introduces the following choice:
> 
> * specify target
>   or
> * specify work_dir and do not specify target

Actually target is path to CMakeList.txt file describes project. You can specify both work_dir and target options

**Comment by rshym** [comment]
Yes, I understand, that we can provide both variables, but I still do not see a reason to add a new variable to provide a path to out-of-tree component.
Please see an example from outdated PR with out-of-tree component:
https://github.com/xen-troops/meta-xt-prod-devel-rcar/pull/60/commits/142c387845d04def5928dc0dd3582e922a2e2544#diff-e78b63e574102e5f392ff67a928a6d7c926768b1cb99351fcf09fe007cb02501R329

It just adds ".." to `target` to work with the out-of-tree applications.

My question is not objection really. I am just not sure if we need to add a variable when we can just use a relative path in yaml-descriptor.

**Comment by dsemenets** [comment]
Your proposal will not work. I agree the 'target' option can be mandatory and we can set this option to ".". But we have to point work directory. In case out-of-tree project we have no CMakeList.txt in 'build_dir'

**Comment by lorc** [comment]
Okay, if I got this correct, `work_dir` points to an application. Should we call this parameter `app_root_dir`?

**Comment by dsemenets** [comment]
I'm not sure. In other builder similar option named 'work_dir'. I don't think we have produce entities...

**Comment by lorc** [comment]
"work_dir" is a dir where build is happening.  At least, this is how it described for Yocto builder. AFAIK, no other builders have this parameter.

Also, I am not sure why "target" option is not working in your case. I can build our dom0 app with the following command: `west build -b xenvm-qemu zephyr-dom0-xt`.  Mouling show generate similar command with `target: zephyr-dom0-xt` option. So, what exactly you are trying to achieve with this PR?

**Review by lorc — COMMENTED** [review]
Could you please confirm that with this change, default location of build dir is `zephyr/build`? This is required for backward compatibility.

File: docs/user-reference.rst [hunk 1]
```diff
@@ -581,7 +582,7 @@ Mandatory options:
 * :code:`type` - builder type. Should be :code:`zephyr` for this type
   of builder.
 
-* :code:`board` - target board name. For example: `xenvm` or `xenvm_gicv3`
+* :code:`board` - target board name. For example: `xenvm` or `xenvm_qemu`
```
[the remark this answers is not in the document]
Comment (dsemenets) [reply docs/user-reference.rst]: Done

**Comment by lorc** [comment]
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

Please confirm that https://github.com/xen-troops/meta-xt-prod-devel-rcar/pull/77 works as intended before we'll merge this.

**Comment by dsemenets** [comment]
I confirm [xen-troops/meta-xt-prod-devel-rcar/pull/77](https://github.com/xen-troops/meta-xt-prod-devel-rcar/pull/77) works

**Comment by lorc** [comment]
Thank you. Could you please add my `Acked-by` to the second patch?

**Pull request merged**